src/main/java/com/google/api/generator/gapic/composer/defaultvalue/DefaultValueComposer.java

```diff
@@ -175,17 +178,46 @@ static Expr createDefaultValue(Field f, boolean useExplicitInitTypeInGenerics) {
   }
 
   public static Expr createDefaultValue(
-      ResourceName resourceName, List<ResourceName> resnames, String fieldOrMessageName) {
-    return createDefaultValueResourceHelper(resourceName, resnames, fieldOrMessageName, true);
+      ResourceName resourceName,
+      boolean isChildType,
+      List<ResourceName> resnames,
+      String fieldOrMessageName) {
+    return createDefaultValueResourceHelper(
+        resourceName, isChildType, resnames, fieldOrMessageName, true);
+  }
+
+  private static Optional<ResourceName> findParentResource(
+      ResourceName childResource, List<ResourceName> resourceNames) {
+    Map<String, ResourceName> patternToResourceName = new HashMap<>();
+
+    for (ResourceName resourceName : resourceNames) {
+      for (String parentPattern : resourceName.patterns()) {
+        patternToResourceName.put(parentPattern, resourceName);
+      }
+    }
+
+    for (String childPattern : childResource.patterns()) {
+      Optional<String> parentPattern = ResourceReferenceUtils.parseParentPattern(childPattern);
```

Comment:
As far as I can tell, `ResourceReferenceUtils.parseParentPattern(childPattern)` is only called by `MethodSigantureParser` and by `Parser`, but only for Google Ads API. So, I don't see a straightforward way to include the mapping to the parent in the `ResourceName` itself, especially since the class is immutable, and you need to have access to all `ResourceName`s before computing the mapping from child to parent.